File: rust/src/model/domain/material.rs
#[derive(Debug, Clone, Copy)]
pub struct Material(pub u32);

impl Material {
    pub fn id(&self) -> String {
        format!("MAT-{:05}", self.0)
    }

    pub fn name(&self) -> String {
        format!("Material {:05}", self.0)
    }
}

File: golang/model/domain/material.go
package domain

import (
	"fmt"
)

type Material struct {
	no int
}

func NewMaterial(no int) Material {
	return Material{no}
}

func (u Material) Id() string {
	return fmt.Sprintf("MAT-%05d", u.no)
}

func (u Material) Name() string {
	return fmt.Sprintf("Material %05d", u.no)
}

func (u Material) No() int {
	return u.no
}
